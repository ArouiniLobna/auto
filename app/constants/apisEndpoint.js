export const apisEndPoints = () => ({
  DUMMY_API: 'https://restcountries.eu/rest/v2',
});
