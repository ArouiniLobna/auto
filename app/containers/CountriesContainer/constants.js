/*
 *
 * CountriesContainer constants
 *
 */

export const LOAD_COUNTRIES =
    'app/CountriesContainer/LOAD_COUNTRIES';
export const LOAD_COUNTRIES_SUCCESS =
    'app/CountriesContainer/LOAD_COUNTRIES_SUCCESS';
export const LOAD_COUNTRIES_ERROR =
    'app/CountriesContainer/LOAD_COUNTRIES_ERROR';
