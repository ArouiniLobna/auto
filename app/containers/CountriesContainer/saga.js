import { all, takeLatest, put, call } from 'redux-saga/effects';
import axios from 'axios';
import { apisEndPoints } from 'constants/apisEndpoint';
import * as types from './constants';
import { loadCountriesSuccess, loadCountriesError } from './actions';

/* **************************************** fetchCountriesLoad start **************************************** */
export function* fetchCountriesLoad() {
	const endpoint = Object.assign({}, apisEndPoints());
	const url = endpoint.DUMMY_API;
	try {
		const globalData = yield call(axios.get, url);
		yield put(loadCountriesSuccess(globalData.data));
	} catch (error) {
		// console.log(error);
		yield put(loadCountriesError(error.response));
	}
}

export function* loadCountries() {
	yield takeLatest(types.LOAD_COUNTRIES, fetchCountriesLoad);
}

// Individual exports for testing
export default function* countriesContainerSaga() {
	// See example in containers/HomePage/saga.js
	yield all([ loadCountries() ]);
}
