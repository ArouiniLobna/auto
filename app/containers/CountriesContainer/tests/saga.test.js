/**
 * Test sagas
 */

import { put, takeLatest } from 'redux-saga/effects';

import { LOAD_COUNTRIES } from '../constants';

import { fetchCountriesLoad, loadCountries } from '../saga';

describe('countriesContainerSaga Saga', () => {
	describe('loadCountries Saga', () => {
		const loadCountriesSaga = loadCountries();

		it('should start task to watch for LOAD_COUNTRIES action', () => {
			const takeLatestDescriptor = loadCountriesSaga.next().value;
			expect(takeLatestDescriptor).toEqual(takeLatest(LOAD_COUNTRIES, fetchCountriesLoad));
		});
	});
});
