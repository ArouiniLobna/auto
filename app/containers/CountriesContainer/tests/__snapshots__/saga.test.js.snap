// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`countriesContainerSaga Saga fetchCountriesLoad Saga should call the loadCountriesError action if the response errors 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      "https://restcountries.eu/rest/v2",
    ],
    "context": null,
    "fn": [Function],
  },
  "type": "CALL",
}
`;

exports[`countriesContainerSaga Saga fetchCountriesLoad Saga should call the loadCountriesError action if the response errors 2`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "action": Object {
      "error": undefined,
      "type": "app/CountriesContainer/LOAD_COUNTRIES_ERROR",
    },
    "channel": undefined,
  },
  "type": "PUT",
}
`;

exports[`countriesContainerSaga Saga fetchCountriesLoad Saga should dispatch the loadCountriesSuccess action if it requests the data successfully 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      "https://restcountries.eu/rest/v2",
    ],
    "context": null,
    "fn": [Function],
  },
  "type": "CALL",
}
`;

exports[`countriesContainerSaga Saga fetchCountriesLoad Saga should dispatch the loadCountriesSuccess action if it requests the data successfully 2`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "action": Object {
      "error": undefined,
      "type": "app/CountriesContainer/LOAD_COUNTRIES_ERROR",
    },
    "channel": undefined,
  },
  "type": "PUT",
}
`;

exports[`countriesContainerSaga Saga loadCountries Saga should dispatch the loadCountriesSuccess action if it requests the data successfully 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      "app/CountriesContainer/LOAD_COUNTRIES",
      [Function],
    ],
    "context": null,
    "fn": [Function],
  },
  "type": "FORK",
}
`;

exports[`countriesContainerSaga Saga loadCountries Saga should dispatch the loadCountriesSuccess action if it requests the data successfully 2`] = `undefined`;
